### PR TITLE
feat: add `math/base/special/nanmin`

### LINKED ISSUE #2318: [RFC]: add `@stdlib/math/base/special/nanmin`
### Description

This RFC proposes adding the package `@stdlib/math/base/special/nanmin`. If one operand is `NaN`, the other operand is always returned; otherwise, perform normal minimum (e.g., by calling `@stdlib/math/base/special/min`).

```javascript
var m = nanmin( 3.0, 4.0 );
// returns 3.0

m = nanmin( NaN, 4.0 );
// returns 4.0

m = nanmin( 3.0, NaN );
// returns 3.0

m = nanmin( NaN, NaN );
// returns NaN
```

Package: `@stdlib/math/base/special/nanmin`
Alias: `nanmin`

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by RidamGarg**
Added Package `@stdlib/math/base/special/nanmin`.

Resolves: https://github.com/stdlib-js/stdlib/issues/2318

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
@RidamGarg Are you planning on adding the C implementation to this PR or in a subsequent PR?

**Comment by kgryte** [comment]
@RidamGarg Looks like you'll also need to resolve some CI failures for linting and benchmarks.

**Comment by RidamGarg** [comment]
@kgryte made some changes. Can you please try again? I am planning to first complete the JS Implementation for both #2318 & #2317. After that I'll start C implementation of both

**Comment by RidamGarg** [comment]
@kgryte I am able to make the changes by cloning the project again. Can you please check once if you can merge PR now?

**Comment by RidamGarg** [comment]
@kgryte made the linting changes again & pushed the code. Can You please tell me? How Can I check these linting errors on my system?

**Comment by kgryte** [comment]
@RidamGarg You'll want to consult the project development guide and leverage various `make` recipes (e.g., https://github.com/stdlib-js/stdlib/tree/develop/tools/make/lib/lint). Since you are running on Windows, you'll likely need to use WSL. But perhaps a more straightforward approach would be to leverage a dev container https://github.com/stdlib-js/stdlib/tree/develop/.devcontainer which sets up an environment for development.

**Comment by RidamGarg** [comment]
Okay @kgryte I'll go through this. Thanks for sharing!

**Comment by RidamGarg** [comment]
@kgryte, Can you please check? If we are able to merge the code now once. As I resolved the required eslint issues. So, that after this I can start my work on another issue?

File: lib/node_modules/@stdlib/math/base/special/nanmin/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# nanmin
+
+> If one operand is NaN, the other operand is always returned; otherwise, perform normal minimum
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/README.md]: Maybe instead:

```suggestion
> Return the minimum value, ignoring NaN.
```

The description should start by explaining what the function does. Also good to aim at being succinct. Notice also that we require a period at the end of the description.

File: lib/node_modules/@stdlib/math/base/special/nanmin/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+```javascript
+var v = nanmin( 4.2, 3.14 );
+// returns 3.14
+
+v = nanmin( +0.0, -0.0 );
+// returns -0.0
+```
+
+If any argument is `NaN`, the function returns `other operand`.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/README.md]:
```suggestion
If any argument is `NaN`, the function returns the other operand.
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+```javascript
+var v = nanmin( 4.2, NaN );
+// returns 4.2
+
+v = nanmin( NaN, 3.14 );
+// returns 3.14
+```
+
+
+If both argument is `NaN`, the function returns `NaN`.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/README.md]:
```suggestion
If both argument are `NaN`, the function returns `NaN`.
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/README.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+<section class="related">
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/math/base/special/max`][@stdlib/math/base/special/max]</span><span class="delimiter">: </span><span class="description">return the maximum value.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/minabs`][@stdlib/math/base/special/minabs]</span><span class="delimiter">: </span><span class="description">return the minimum absolute value.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/minn`][@stdlib/math/base/special/minn]</span><span class="delimiter">: </span><span class="description">return the minimum value.</span>
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/README.md]: We auto-populate these sections from metadata, so this should be left empty.

```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/README.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+<section class="links">
+
+<!-- <related-links> -->
+
+[@stdlib/math/base/special/max]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/max
+
+[@stdlib/math/base/special/minabs]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/minabs
+
+[@stdlib/math/base/special/minn]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/minn
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/README.md]: Should be left empty as auto-populated.

```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/benchmark/benchmark.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var values;
+	var x;
+	var y;
+	var z;
+	var i;
+
+	values = [3, 2, 2.0, 1.5, NaN];
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/benchmark/benchmark.js]:
```suggestion
	values = [ 3, 2, 2.0, 1.5, NaN ];
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/test/test.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+tape( 'the function returns `NaN` if both operands are `NaN`', function test( t ) {
+	var v;
+
+	v = nanmin( NaN, NaN );
+	t.strictEqual( isnan( v ), true, 'returns NaN value' );
+	t.end();
+});
+
+tape( 'the function returns `Not NaN Value` if one of the operand is `NaN`', function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/test/test.js]:
```suggestion
tape( 'the function returns the non-NaN value if one of the operands is `NaN`', function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/lib/main.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+	if (isnan(x) && isnan(y)) {
+		return NaN;
+	}
+	if (isnan(x)) {
+		return y;
+	}
+	if (isnan(y)) {
+		return x;
+	}
+	return min(x, y);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/lib/main.js]: We can remove some redundancy by doing:

```suggestion
	if ( isnan( x ) ) { 
		return ( isnan( y ) ? NaN : y ); 
	} else {
		return ( isnan( y ) ? x : min( x, y ) );
	}
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/docs/types/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/**
+* If one operand is NaN, the other operand is always returned; otherwise, perform normal minimum.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/docs/types/index.d.ts]:
```suggestion
* Returns the minimum value, ignoring NaN.
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/lib/index.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* If one operand is NaN, the other operand is always returned; otherwise, perform normal minimum.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/lib/index.js]:
```suggestion
* Return the minimum value, ignoring NaN.
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/lib/main.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+// MODULES //
+
+var min = require( '@stdlib/math/base/special/min');
+var isnan = require( '@stdlib/math/base/assert/is-nan');
+
+
+// MAIN //
+
+/**
+* If one operand is NaN, the other operand is always returned; otherwise, perform normal minimum.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/lib/main.js]:
```suggestion
* Returns the minimum value, ignoring NaN.
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/package.json [hunk 12]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@stdlib/math/base/special/nanmin",
+  "version": "0.0.0",
+  "description": "If one operand is NaN, the other operand is always returned; otherwise, perform normal minimum",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/package.json]:
```suggestion
  "description": "Return the minimum value, ignoring NaN.",
```

**Review by Planeshifter — COMMENTED** [review]
Thanks @RidamGarg for your PR! Looks good overall; left a few suggestions, the main relating to the package description.

**Comment by RidamGarg** [comment]
@Planeshifter made the required changes

File: lib/node_modules/@stdlib/math/base/special/nanmin/lib/main.js [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+*
+* @example
+* var v = nanmin( NaN, NaN);
+* // returns NaN
+*/
+function nanmin( x, y ) {
+	if (isnan(x)) {
+		return (( isnan(y) ) ? NaN : y);
+	}
+	return (( isnan(y) ) ? x : min(x, y));
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmin/lib/main.js]:
```suggestion
	if ( isnan( x ) ) {
		return ( isnan(y) ) ? NaN : y;
	}
	return ( isnan( y ) ) ? x : min( x, y );
```

**Comment by Planeshifter** [comment]
Thank you for making the changes and your contribution, @RidamGarg. It is much appreciated!

**Comment by Planeshifter** [comment]
@RidamGarg Could you please tick the checkbox that you read and accept the contributing guidelines, which is present in the PR template and which I inserted into your opening comment? Will merge your PR afterwards. Thank you!

**Comment by RidamGarg** [comment]
Done @Planeshifter

File: lib/node_modules/@stdlib/math/base/special/nanmin/README.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+```javascript
+var v = nanmin( 4.2, NaN );
+// returns 4.2
+
+v = nanmin( NaN, 3.14 );
+// returns 3.14
+```
+
+
+If both argument are `NaN`, the function returns `NaN`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/nanmin/README.md:66]:
```suggestion
If both arguments are `NaN`, the function returns `NaN`.
```

File: lib/node_modules/@stdlib/math/base/special/nanmin/examples/index.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+var nanmin = require( './../lib' );
+
+var m = nanmin( 3.0, 4.0 );
+// returns 3.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/nanmin/examples/index.js:24]: We should be logging each return value to the console.

File: lib/node_modules/@stdlib/math/base/special/nanmin/lib/main.js [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+* var v = nanmin( 4.14, NaN );
+* // returns 4.14
+*
+* @example
+* var v = nanmin( NaN, NaN);
+* // returns NaN
+*/
+function nanmin( x, y ) {
+	if ( isnan( x ) ) {
+		return ( isnan(y) ) ? NaN : y;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/nanmin/lib/main.js:50]:
```suggestion
		return ( isnan( y ) ) ? NaN : y;
```

Consistent spacing.

File: lib/node_modules/@stdlib/math/base/special/nanmin/README.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+```javascript
+var v = nanmin( NaN, NaN );
+// returns NaN
+
+```
+
+</section>
+
+<!-- /.usage -->
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/nanmin/README.md:77]: This readme is missing various sections (eg, examples, links, etc).

**Comment by kgryte** [comment]
@RidamGarg You'll want to address the small comments in a follow-up PR.

cc @Planeshifter

File: lib/node_modules/@stdlib/math/base/special/nanmin/README.md (hunk 17 above)
Comment (RidamGarg) [reply lib/node_modules/@stdlib/math/base/special/nanmin/README.md:77]: I added the section previously but got this comment from @Planeshifter. We auto-populate these sections from metadata, so this should be left empty. Should I add this in my next PR?

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/nanmin/README.md:77]: @RidamGarg Sorry if my comment was not fully clear; the suggested change was to remove the inner content of the sections, not remove them altogether. 

So 

```md
<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

<section class="related">

* * *

## See Also

-   <span class="package-name">[`@stdlib/math/base/special/max`][@stdlib/math/base/special/max]</span><span class="delimiter">: </span><span class="description">return the maximum value.</span>
-   <span class="package-name">[`@stdlib/math/base/special/minabs`][@stdlib/math/base/special/minabs]</span><span class="delimiter">: </span><span class="description">return the minimum absolute value.</span>
-   <span class="package-name">[`@stdlib/math/base/special/minn`][@stdlib/math/base/special/minn]</span><span class="delimiter">: </span><span class="description">return the minimum value.</span>

</section>

<!-- /.related -->

<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="links">

<!-- <related-links> -->

[@stdlib/math/base/special/max]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/max

[@stdlib/math/base/special/minabs]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/minabs

[@stdlib/math/base/special/minn]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/minn

<!-- </related-links> -->

</section>

<!-- /.links -->
```

should have become

```md
<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

<section class="related">

</section>

<!-- /.related -->

<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="links">

<!-- <related-links> -->

<!-- </related-links> -->

</section>

<!-- /.links -->
```

We auto-populate the contents for the related packages, but that will only work if the sections are there.

Notice that the examples section is separate from that. It should contain the example from `examples/index.js` file; see other packages for reference.

Comment (RidamGarg) [reply lib/node_modules/@stdlib/math/base/special/nanmin/README.md:77]: Ooh, Okay @Planeshifter, Now I got it. Thanks for the clarification.

**Pull request merged**